dawn/renderer/utils/lcm_ws_bridge.js

```diff
@@ -0,0 +1,148 @@
+/*
+  LCM WebSocket Bridge: javascript client library
+
+  This library imitates the LCM APIs, and fulfills requests by forwarding them
+  to a websocket server.
+
+  PUBLIC API:
+
+    new LCM(websocket_url)
+    on_ready(callback)
+    subscribe(channel, msg_type, callback)
+    unsubscribe(subscription_id)
+    publish(channel, msg)
+
+  EXAMPLE CODE:
+
+    let l = new LCM("ws://localhost:8000");
+
+    l.on_ready(function() {
+      let sub = l.subscribe("sprocket/health", "health", function(msg) {
+        alert(msg.header.seq);
+      });
+
+      window.setTimeout(function() {
+        l.unsubscribe(sub);
+      }, 1000);
+
+      l.publish("/forest/cmd", {
+        __type__: "forest_cmd",
+        header: {
+          __type__: "header",
+          seq: 5,
+          time: 0
+        },
+        lights: [[true, true, false],
+                 [true, true, false],
+                 [true, true, false],
+                 [true, true, false],
+                 [true, true, false],
+                 [true, true, false],
+                 [true, true, false],
+                 [true, true, false]],
+        servos: [0, 0, 0, 0, 0, 0, 0, 0]
+      });
+    });
+*/
+import WebSocket from 'ws';
+
+
+const CircularJSON = require('circular-json');
```

Comment:
Would you perhaps be willing to spend some time to get to the bottom of this issue? Maybe it would be possible to leave a comment explaining why this is required, or even to get rid of whatever is creating the circular objects in the first place.

The serialization issue you're describing sounds very surprising to me, and if you no longer remember the details then I'm not sure anyone in PiE understands why this code works. I'm of the opinion that it's better to spend time once to figure out what's going on, rather than punting the rediscovery of this issue down to new staff at some point in the future.